### ISSUE TITLE
[Jetson Xavier NX] Perception module build issue

### ISSUE THREAD

**Issue by smartview-ai** [ApolloAuto/apollo#13298]
While building Perception module on "Jetson Xavier NX" getting SSE related error as below,
**_error: unrecognized command line option ‘-msse4.1’_**
to resolve this **disabled** SSE and rebuild the PCL.
unfortunately, problem didn't resolve causing another error, 
**_error: '__m128i' was not declared in this scope_**
![perception_build_err](https://user-images.githubusercontent.com/71485769/104091348-2e5d5880-52c0-11eb-9d6d-86d01610b2eb.png)

any suggestions are greatly appreciated.

**Comment by muzi2045** [comment]
it looks like the problem of ground Detector in Perception Module.
it using SSE2 with X86 arch and those inline assemble code should be translate into ARM arch, like sse2neon.

**Comment by smartview-ai** [comment]
yes, you were right. SSE option ( _copts = ["-msse4.1"]_ ) was included in ground_detector.
_modules/perception/lidar/lib/ground_detector/spatio_temporal_ground_detector/BUILD_
would you mind provide me little bit more info on translating into ARM arch

**Comment by storypku** [comment]
> yes, you were right. SSE option ( _copts = ["-msse4.1"]_ ) was included in ground_detector.
> _modules/perception/lidar/lib/ground_detector/spatio_temporal_ground_detector/BUILD_
> would you mind provide me little bit more info on translating into ARM arch

Sorry that Apollo hasn't support fully the aarch64 architecture.  For SIMD instructions on Arm NEON architecture,
Please refer to https://developer.arm.com/architectures/instruction-sets/simd-isas/neon, or the sse2neon project @muzi2045 just mentioned: https://github.com/DLTcollab/sse2neon

**Comment by smartview-ai** [comment]
Thank you for yours prompt response.

**Comment by smartview-ai** [comment]
> it looks like the problem of ground Detector in Perception Module.
> it using SSE2 with X86 arch and those inline assemble code should be translate into ARM arch, like sse2neon.
@muzi2045, thank you.

**Comment by RoseBrother** [comment]
> While building Perception module on "Jetson Xavier NX" getting SSE related error as below,
> **_error: unrecognized command line option ‘-msse4.1’_**
> to resolve this **disabled** SSE and rebuild the PCL.
> unfortunately, problem didn't resolve causing another error,
> **_error: '__m128i' was not declared in this scope_**
> 
> any suggestions are greatly appreciated.
> 
> ![perception_build_err](https://user-images.githubusercontent.com/71485769/104091348-2e5d5880-52c0-11eb-9d6d-86d01610b2eb.png)


Have you found a solution yet? I have the same problem as you.

**Comment by xhzzhang** [comment]
Hi, did you have success build apollo-8.0 on jetson NX device, I got link fail paddle libs for perception,and tried to build paddle source codes refer to official steps, but got many error.
in addition, when I run canbus module, got below message:

[t@in-dev-docker:/apollo]$ cyber_launch start modules/canbus/launch/canbus.launch
[cyber_launch_12062] INFO Launch file [/apollo/modules/canbus/launch/canbus.launch]
[cyber_launch_12062] INFO ========================================================================================================================
[cyber_launch_12062] INFO Load module [canbus] library: [canbus] [CYBER_DEFAULT] conf: [/apollo/modules/canbus/dag/canbus.dag] exception_handler: []
[cyber_launch_12062] INFO Start process [canbus] successfully. pid: 12069
[cyber_launch_12062] INFO ------------------------------------------------------------------------------------------------------------------------
[canbus ]  WARNING: Logging before InitGoogleLogging() is written to STDERR
[canbus ]  I0620 10:27:38.160516 12069 module_argument.cc:81] []command: mainboard -d /apollo/modules/canbus/dag/canbus.dag -p canbus -s CYBER_DEFAULT 
[canbus ]  I0620 10:27:38.161417 12069 global_data.cc:153] []host ip: 192.168.1.6
[canbus ]  I0620 10:27:38.164485 12069 module_argument.cc:57] []binary_name_ is mainboard, process_group_ is canbus, has 1 dag conf
[canbus ]  I0620 10:27:38.164532 12069 module_argument.cc:60] []dag_conf: /apollo/modules/canbus/dag/canbus.dag
[canbus ]  E0620 10:27:38.290521 12069 canbus_component.cc:52] [canbus]/opt/apollo/neo/packages/canbus-vehicle-lincoln-dev/latest/lib/liblincoln_vehicle_factory_lib.so No such vehicle library
[canbus ]  E0620 10:27:38.291342 12069 module_controller.cc:67] [mainboard]Failed to load module: /apollo/modules/canbus/dag/canbus.dag
[canbus ]  E0620 10:27:38.291414 12069 mainboard.cc:39] [mainboard]module start error.
[canbus ]  
[cyber_launch_12062] ERROR Process [canbus] has died [pid 12069, exit code 255, cmd mainboard -d /apollo/modules/canbus/dag/canbus.dag -p canbus -s CYBER_DEFAULT].
[cyber_launch_12062] INFO All processes has died.
[cyber_launch_12062] INFO Cyber exit.
[cyber_launch_12062] INFO All processes have been stopped.

Is someone can help me? great thanks!

**Comment by HandsomeAIccx** [comment]
> Hi, did you have success build apollo-8.0 on jetson NX device, I got link fail paddle libs for perception,and tried to build paddle source codes refer to official steps, but got many error. in addition, when I run canbus module, got below message:
> 
> [t@in-dev-docker:/apollo]$ cyber_launch start modules/canbus/launch/canbus.launch [cyber_launch_12062] INFO Launch file [/apollo/modules/canbus/launch/canbus.launch] [cyber_launch_12062] INFO ======================================================================================================================== [cyber_launch_12062] INFO Load module [canbus] library: [canbus] [CYBER_DEFAULT] conf: [/apollo/modules/canbus/dag/canbus.dag] exception_handler: [] [cyber_launch_12062] INFO Start process [canbus] successfully. pid: 12069 [cyber_launch_12062] INFO ------------------------------------------------------------------------------------------------------------------------ [canbus ] WARNING: Logging before InitGoogleLogging() is written to STDERR [canbus ] I0620 10:27:38.160516 12069 module_argument.cc:81] []command: mainboard -d /apollo/modules/canbus/dag/canbus.dag -p canbus -s CYBER_DEFAULT [canbus ] I0620 10:27:38.161417 12069 global_data.cc:153] []host ip: 192.168.1.6 [canbus ] I0620 10:27:38.164485 12069 module_argument.cc:57] []binary_name_ is mainboard, process_group_ is canbus, has 1 dag conf [canbus ] I0620 10:27:38.164532 12069 module_argument.cc:60] []dag_conf: /apollo/modules/canbus/dag/canbus.dag [canbus ] E0620 10:27:38.290521 12069 canbus_component.cc:52] [canbus]/opt/apollo/neo/packages/canbus-vehicle-lincoln-dev/latest/lib/liblincoln_vehicle_factory_lib.so No such vehicle library [canbus ] E0620 10:27:38.291342 12069 module_controller.cc:67] [mainboard]Failed to load module: /apollo/modules/canbus/dag/canbus.dag [canbus ] E0620 10:27:38.291414 12069 mainboard.cc:39] [mainboard]module start error. [canbus ] [cyber_launch_12062] ERROR Process [canbus] has died [pid 12069, exit code 255, cmd mainboard -d /apollo/modules/canbus/dag/canbus.dag -p canbus -s CYBER_DEFAULT]. [cyber_launch_12062] INFO All processes has died. [cyber_launch_12062] INFO Cyber exit. [cyber_launch_12062] INFO All processes have been stopped.
> 
> Is someone can help me? great thanks!

Hi. I also encountered the same problem. Have you solved it? Could you help me?
